File: src/main/java/me/miki/shindo/management/cosmetic/wing/WingCategory.java
package me.miki.shindo.management.cosmetic.wing;

import lombok.Getter;
import me.miki.shindo.management.language.TranslateText;
import me.miki.shindo.utils.animation.ColorAnimation;
import me.miki.shindo.utils.animation.simple.SimpleAnimation;

@Getter
public enum WingCategory {
    ALL(TranslateText.ALL.getText()), SHINDO("Shindo"), LUNAR("Lunar");

    private final String name;
    private final SimpleAnimation backgroundAnimation = new SimpleAnimation();
    private final ColorAnimation textColorAnimation = new ColorAnimation();

    WingCategory(String name) {
        this.name = name;
    }

}

File: src/main/kotlin/me/miki/shindo/management/cosmetic/wing/WingCategory.kt
package me.miki.shindo.management.cosmetic.wing

import me.miki.shindo.management.language.TranslateText
import me.miki.shindo.ui.animation.v2.value.ColorAnimation
import me.miki.shindo.ui.animation.v2.value.SimpleAnimation

enum class WingCategory(private val names: String) {
    ALL(TranslateText.ALL.getText()),
    CLASSIC("Classic"),
    FANTASY("Fantasy"),
    TECH("Tech");

    private val backgroundAnimation = SimpleAnimation()
    private val textColorAnimation = ColorAnimation()

    fun getName(): String = names
    fun getBackgroundAnimation(): SimpleAnimation = backgroundAnimation
    fun getTextColorAnimation(): ColorAnimation = textColorAnimation
}
